### PR TITLE
fix: call the counterpart function of availabilitySet when the instance is not a vmss vm

### LINKED ISSUE #596: Ensuring LB fails with ListAgentPoolLBs: failed to get agent pool vmSet names
**What happened**: 

Updated cloud-provider to v0.7.3 and CAPZ test E2E fail with this error:
```
I0419 22:08:22.703254       1 controller.go:368] Ensuring load balancer for service default/web-ilb
I0419 22:08:22.703385       1 azure_loadbalancer.go:1429] reconcileLoadBalancer for service(default/web-ilb) - wantLb(true): started
I0419 22:08:22.703631       1 event.go:291] "Event occurred" object="default/web-ilb" kind="Service" apiVersion="v1" type="Normal" reason="EnsuringLoadBalancer" message="Ensuring load balancer"
I0419 22:08:22.748645       1 azure_backoff.go:311] LoadBalancerClient.List(capz-e2e-7hbnku) success
E0419 22:08:22.748769       1 azure_loadbalancer.go:1433] reconcileLoadBalancer: failed to reconcile shared load balancer: %!w(*fmt.wrapError=&{reconcileSharedLoadBalancer: failed to list LB: ListAgentPoolLBs: failed to get agent pool vmSet names: not a vmss instance 0xc0006213c0})
E0419 22:08:22.748815       1 azure_loadbalancer.go:104] reconcileLoadBalancer(default/web-ilb) failed: reconcileSharedLoadBalancer: failed to list LB: ListAgentPoolLBs: failed to get agent pool vmSet names: not a vmss instance
E0419 22:08:22.748934       1 controller.go:275] error processing service default/web-ilb (will retry): failed to ensure load balancer: reconcileSharedLoadBalancer: failed to list LB: ListAgentPoolLBs: failed to get agent pool vmSet names: not a vmss instance
I0419 22:08:22.749407       1 event.go:291] "Event occurred" object="default/web-ilb" kind="Service" apiVersion="v1" type="Warning" reason="SyncLoadBalancerFailed" message="Error syncing load balancer: failed to ensure load balancer: reconcileSharedLoadBalancer: failed to list LB: ListAgentPoolLBs: failed to get agent pool vmSet names: not a vmss instance"
```

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1323/pull-cluster-api-provider-azure-e2e-full/1384253528750428160/artifacts/clusters/capz-e2e-7hbnku/kube-system/cloud-controller-manager/cloud-controller-manager.log

This seems to be a regression from #588. This is on a cluster without the multiple SLBs feature.

**What you expected to happen**: Both VMAS and VMSS nodes should be supported for clusters in "vmss" vmType with Standard LB, as stated in https://github.com/kubernetes-sigs/cloud-provider-azure/issues/338#issuecomment-642076028.

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #596

**Special notes for your reviewer**:

**Release note**:
```
fix: call the counterpart function of availabilitySet when the instance is not a vmss vm
```

### CODE REVIEW

File: pkg/provider/azure_vmss.go [hunk 1]
```diff
@@ -1674,5 +1674,10 @@ func (ss *ScaleSet) EnsureBackendPoolDeletedFromVMSets(vmssNamesMap map[string]b
 
 // GetAgentPoolVMSetNames returns all VMSS names according to the nodes
 func (ss *ScaleSet) GetAgentPoolVMSetNames(nodes []*v1.Node) (*[]string, error) {
-	return ss.getAgentPoolScaleSets(nodes)
+	vmssNames, err := ss.getAgentPoolScaleSets(nodes)
```
Comment (feiskyer) [inline pkg/provider/azure_vmss.go:1677]: we probably need to get vmSet list node by node? there're chances some nodes are VMSS, while others are VMAS

**Comment by CecileRobertMichon** [comment]
/hold

tested the changes in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1323, it's failing with:

```
2021-04-20 23:24:11 +0000 UTC  Warning  SyncLoadBalancerFailed  service/web-ilb  Error syncing load balancer: failed to ensure load balancer: reconcileSharedLoadBalancer: failed to list LB: ListAgentPoolLBs: failed to get agent pool vmSet names: GetAgentPoolVMSetNames: failed to execute getAgentPoolAvailabilitySets: node (capz-e2e-kemtep-md-0-zxfxj) - has no availability sets
    2021-04-20 23:22:51 +0000 UTC  Warning  SyncLoadBalancerFailed  service/web-ilb  Error syncing load balancer: failed to ensure load balancer: reconcileSharedLoadBalancer: failed to list LB: ListAgentPoolLBs: failed to get agent pool vmSet names: GetAgentPoolVMSetNames: failed to execute getAgentPoolAvailabilitySets: node (capz-e2e-kemtep-md-0-dt68h) - has no availability sets
   ```

**Comment by CecileRobertMichon** [comment]
/hold cancel

Verified this fixes the CAPZ scenario here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1323

Here are the CCM logs for that PR in case you want to verify everything is working as expected https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1323/pull-cluster-api-provider-azure-e2e-full/1384945821480390656/artifacts/clusters/capz-e2e-y0d1op/kube-system/cloud-controller-manager/cloud-controller-manager.log

**Comment by nilo19** [comment]
@CecileRobertMichon I think it's expected since there's no error in the log.

**Review by feiskyer — APPROVED** [review]
/lgtm

**Pull request merged**